### PR TITLE
Use kpromo to promote images for releases

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What type of PR is this?**

**What this PR does / why we need it**: THIS WILL OPEN A PR FOR YOU IN k8s.io!! no more searching through the UI, copy pasting SHAs, all it takes is running "make promote-images" and voila! It will even assign the right people automatically 🪄 

This is based on some great work by @puerco +  @palnabarun in krel and @justaugustus in https://github.com/kubernetes-sigs/promo-tools/pull/458 ❤️

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by CecileRobertMichon** [comment]
/cc @vincepri

PTAL, I'm planning on opening a PR in CAPI as well

**Review by vincepri — APPROVED** [review]
This is amazing ❤️

**Comment by puerco** [comment]
Great work Cecile! Nice to see this in use!

File: hack/get-project-maintainers.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KEYS=()
+while IFS='' read -r line; do KEYS+=("$line"); done < <(yq -r '.aliases["cluster-api-azure-maintainers"][]' < OWNERS_ALIASES)
```
Comment (CecileRobertMichon) [inline hack/get-project-maintainers.sh:22]: todo: use YQ in hack/tools

File: hack/get-project-maintainers.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/.. 
+cd "${REPO_ROOT}"
+
+YQ=$(realpath hack/tools/bin/yq)
+make "${YQ}" &>/dev/null
```
Comment (mboersma) [inline hack/get-project-maintainers.sh:25]: Calling `make` from the script that was invoked to populate a Makefile variable... 🤯

**Review by mboersma — APPROVED** [review]
/lgtm

I tried it out locally (without setting `GITHUB_TOKEN`) and it seems to work as advertised.

File: hack/get-project-maintainers.sh (hunk 2 above)
Comment (CecileRobertMichon) [reply hack/get-project-maintainers.sh:25]: I could have also put it as a make target dependency but I figured the script should work even when it's not called from that make target so it's better to install the dependency directly in the script

**Comment by CecileRobertMichon** [comment]
example PR made from this branch: https://github.com/kubernetes/k8s.io/pull/3129

**Comment by CecileRobertMichon** [comment]
@mboersma could I get another lgtm? I pushed an update to add quotes around the reviewers https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1877/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R521

**Comment by mboersma** [comment]
Looks good, but do you want to squash the commits first?

**Comment by CecileRobertMichon** [comment]
I kept them separate since they are technically two different things (one is adding a net new target and tool, the other is bumping an existing tool), but I can squash if you think it's not distinct enough to warrant two commits

**Comment by mboersma** [comment]
/lgtm

No, I agree they are logically separate (and easier to review). I thought we needed to squash manually, but I'm probably wrong. :-)

**Review by devigned — APPROVED** [review]
/lgtm
/approve

**Pull request merged**